java-checks/src/main/java/org/sonar/java/checks/ThreadRunCheck.java

```diff
@@ -50,7 +54,19 @@ protected void onMethodInvocationFound(MethodInvocationTree mit) {
     if (parent != null && THREAD_RUN_METHOD_MATCHER.matches((MethodTree) parent)) {
       return;
     }
-    reportIssue(ExpressionUtils.methodName(mit), "Call the method Thread.start() to execute the content of the run() method in a dedicated thread.");
+
+    IdentifierTree methodName = ExpressionUtils.methodName(mit);
```

Comment:
I don't see a reason to be specific about the type of `methodName`, so maybe we could store it as a simple `Tree` and also change the `computeQuickFix` to take a `Tree` as parameter
```suggestion
    Tree methodName = ExpressionUtils.methodName(mit);
```

Reply:
But what does it buy to be more generic here?
Whereas if we keep `IdentifierTree`, assuming that in the future there is a change and the change requires a call (from `methodName) to a method defined in `IdentifierTree`, we would not have to cast.
What do you think?
In general, I think it is better to be more specific.

Reply:
Mh, yeah it's a good point, since `.methodName` returns a `IdentifierTree` and it doesn't have any cost to just keep it as it is, being specific doesn't hurt and it could be useful in the future, I agree